File: 프로그래머스/lv3/지준혁/정수 삼각형.cpp
// 작성 중 시간...
class Solution {
    
    int dp[504][504];
    public int solution(int[][] triangle) {
        
        dp[0][0] = triangle[0][0];
        for (int i = 1; i <= n; ++i) {
            for (int j = 0; j <= i; ++j) {
                if (j == 0) dp[i][j] = dp[i - 1][j] + triangle[i][j];
                else if (j == i) dp[i][j] = dp[i - 1][j - 1] + triangle[i][j];
                else 
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv3/이선영/정수 삼각형.java
import java.util.*;
class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        
        for(int i=triangle.length-2; i>=0; i--){
            for(int j=0; j<triangle[i].length; j++){
                int left = triangle[i+1][j];
                int right = triangle[i+1][j+1];
                triangle[i][j] += Math.max(left, right);
            }
        }
        answer = triangle[0][0];
        return answer;
    }
}
